src/test/resources/update/formatting_change/new.java

```diff
@@ -0,0 +1,14 @@
+// ToDo: Preserve the format of the code.
```

Comment:
@slarse Shouldn't we instead ignore the formatting because we intend to keep the AST diff and that is how we will minimise line-based diff?

For example, consider https://github.com/algomaster99/diffmin-examples/pull/8. If we show the following diff to the users,
```diff
class SpacesInLocalVariable {
    public static void main(String[] args) {
-         int x = 1;
+         int y = 1;
    }
}
```
instead of what is shown [here](https://github.com/algomaster99/diffmin-examples/pull/8/files), don't you think this is more minimised and in fact, does achieve the goal of removing spurious formatting changes?

Reply:
It depends. In general it's probably better to preserve formatting. DJPP doesn't print particularly nicely formatted code, and often prints code that simply isn't acceptable to a developer (extra parentheses, extremely long lines, etc).

You could experiment with it, but my guess is that if you don't preserve formatting, you'll pretty much never make a diff any easier to read, or with fewer characters (although there may be fewer lines, as Spoon DJPP doesn't line-break long expressions).

Reply:
@slarse I had a meeting with @monperrus and @khaes-kth today. For this particular example, we agreed that we should not include the spaces because we think it is a "spurious formatting change". Now what exactly is a spurious formatting change is an open question and I think it will be clearer when we see more practical examples. I found one practical example, but it is on Scala. You can have a look at it [here](https://github.com/algomaster99/diffmin-examples/issues/10).

Martin also said that we want something like a _sniperised diff_, whose precise definition is also not clear. However, that diff will have lesser line diff operations compared to what GitHub UI shows. To what extent it will be lowered, depends upon what all we consider to be a spurious formatting change.